### PR TITLE
docs: rewrite reducers guide

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/store/index.md [hunk 1]
```diff
@@ -1,49 +1,63 @@
...
-yarn add @ngrx/store
-```
+The following tutorial shows you how to manage the state of a counter, and how to select and display it within an Angular component.
 
-## Example
+1.  Generate a <a href="https://stackblitz.com/fork/ngrx-start" target="_blank">new project</a> using StackBlitz.
 
-Create a reducer function for each data type you have in your application. The
-combination of these reducers will make up your application state:
+2.  Create an actions file to describe the counter actions to increment, decrement, and reset its value.
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/index.md]: counteractions has a different meaning. Maybe counter events?

File: projects/ngrx.io/content/guide/store/index.md [hunk 2]
```diff
@@ -52,29 +66,35 @@ export function counterReducer(state: number = initialState, action: Action) {
...
 
 ```ts
+// my-counter.component.ts
 import { Component } from '@angular/core';
 import { Store, select } from '@ngrx/store';
 import { Observable } from 'rxjs';
-import { INCREMENT, DECREMENT, RESET } from './counter';
+import { Increment, Decrement, Reset } from '../counter.actions';
 
 interface AppState {
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/index.md]: I'll just inline it in the `Store<{ count: number }>`. I want to add an architecture guide in the future that talks about structure and defining interfaces.

File: projects/ngrx.io/content/guide/store/index.md [hunk 3]
```diff
@@ -98,15 +120,34 @@ export class MyCounterComponent {
...
   }
 
   reset() {
-    this.store.dispatch({ type: RESET });
+    this.store.dispatch(new Reset());
   }
 }
 ```
+
+7.  Add the `MyCounter` component to your `AppComponent` template.
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/index.md]: StackBlitz does this for you when you create the component.

File: projects/ngrx.io/content/guide/store/index.md [hunk 4]
```diff
@@ -52,29 +66,35 @@ export function counterReducer(state: number = initialState, action: Action) {
...
-`select` operator to _select_ slice(s) of state:
+6.  Right click on the `app` folder in StackBlitz and create a new `Component` named `my-counter`. Inject the `Store` service into your component to dispatch the counter actions, and use `select` operator to _select_ slice(s) of state.
 
 ```ts
+// my-counter.component.ts
 import { Component } from '@angular/core';
 import { Store, select } from '@ngrx/store';
 import { Observable } from 'rxjs';
-import { INCREMENT, DECREMENT, RESET } from './counter';
+import { Increment, Decrement, Reset } from '../counter.actions';
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/index.md]: StackBlitz creates the component in a folder. The CLI would create the component in its own folder also.

File: projects/ngrx.io/content/guide/store/index.md [hunk 5]
```diff
@@ -1,49 +1,63 @@
...
-Store is RxJS powered state management for Angular applications, inspired by Redux. Store is a controlled state container designed to help write performant,consistent applications on top of Angular.
+Store is RxJS powered state management for Angular applications, inspired by Redux. Store is a controlled state container designed to help write performant, consistent applications on top of Angular.
 
-## Core Principles
+## Key concepts
 
-- State is a single, immutable data structure that emphasizes type-safety, serializability, and testability.
-- [Actions](guide/store/actions) describe unique events that are dispatched to trigger state changes and/or side effects.
+- [Actions](guide/store/actions) describe unique events that are dispatched from components and services.
 - State changes are handled by pure functions called [reducers](guide/store/reducers) that take the previous state and the next action to compute a new state.
```
[the remark this answers is not in the document]
Comment (brandonroberts) [reply projects/ngrx.io/content/guide/store/index.md]: Good catch

File: projects/ngrx.io/content/guide/store/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/store/index.md]: No, I was wrong. 
I think counter actions is good here.

File: projects/ngrx.io/content/guide/store/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/store/index.md]: TIL "generate component" 😅

File: projects/ngrx.io/content/guide/store/reducers.md [hunk 6]
```diff
@@ -118,42 +117,101 @@ export function reducer(
...
 
-Registering states with `StoreModule.forRoot` ensures that the states are defined upon application startup. In general, you register root states that are alway need to be available to all areas of your application immediately.
+Registering states with `StoreModule.forRoot()` ensures that the states are defined upon application startup. In general, you register root states that always need to be available to all areas of your application immediately.
 
 ## Register feature state
 
-You define
+Feature states behave in the same way root states do, but allow you to define them with specific feature areas in your application. Your state is one large object, and feature states register additional keys and values in that object.
+
+Looking an at example state object, you see how a feature state allows your state to be built up incrementally. Let's start with an empty state object.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/store/reducers.md]: Looking *at an* example state object

**Pull request merged**